publish.go

```diff
@@ -37,7 +38,10 @@ func NewPublisher(ctx context.Context, dataStore datastore.Batching, host host.H
 	tp := gstransport.NewTransport(host.ID(), gs)
 	dtNet := dtnetwork.NewFromLibp2pHost(host)
 
-	tmpDir := os.TempDir()
+	tmpDir, err := ioutil.TempDir("", "golegs-pub")
```

Comment:
Would it make sense to parameterise the tmpDir and fall back onto some default if unspecified?

The rationale is that during testing a temp dir could be set that's created via `testing.T`

Reply:
Yes.  Is it OK to make a breaking interface change (new parameter to New...) or should there be a separate New... that takes an additional parameter, or change New... to take config struct or variadic options?

Reply:
mm... good question. Since parameterisation of `tmpDir`is unrelated to the fix I'd say let's leave a `TODO` and defer to a separate PR

Reply:
Done.  Added TODO.